drivers/include/periph/qdec.h

```diff
@@ -0,0 +1,191 @@
+ *
+ * These signals are decoded to produce a count up or a count down.
+ * On rising or falling edge of one signal, other signal state is checked to
+ * determine direction.
+ *
+ *  1. Rising edge on A and signal B is up       => (C)  => increment counter
+ *  2. Rising edge on A and signal B is down     => (CC) => decrement counter
+ *  3. Falling edge on A and signal B is up      => (C)  => decrement counter
+ *  4. Falling edge on A and signal B is down    => (CC) => increment counter
+ *  5. Rising edge on B and signal A is up       => (C)  => decrement counter
+ *  6. Rising edge on B and signal A is down     => (CC) => increment counter
+ *  7. Falling edge on B and signal A is up      => (C)  => increment counter
+ *  8. Falling edge on B and signal A is down    => (CC) => decrement counter
+ *
+ * According to these cases, three modes are available :
+ *  X1 mode : signal A, rising edges (cases 1 and 2)
+ *  X2 mode : signal A, rising and falling edges (cases 1, 2, 3 and 4)
+ *  X4 mode : signals A and B, rising and falling edges (all cases)
+ *
+ * The mapping/configuration of QDEC devices (timers) and the used pins has to be
+ * done in the board configuration (the board's `periph_conf.h).
+ *
+ * When using the QDEC interface, first thing you have to do is initialize the
+ * QDEC device with the targeted mode. Once the device is initialized,
+ * it will start counting quadrature signals on all configured pins
+ * immediately.
+ *
+ * @{
+ * @file
+ * @brief       Low-level QDEC peripheral driver interface definitions
+ *
+ * @author      Gilles DOFFE <gdoffe@gmail.com>
+ */
+
+#ifndef PERIPH_QDEC_H
+#define PERIPH_QDEC_H
+
+#include <stdint.h>
+#include <limits.h>
+
+#include "periph_cpu.h"
+#include "periph_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Default QDEC access macro
+ */
+#ifndef QDEC_DEV
+#define QDEC_DEV(x)          (x)
+#endif
+
+/**
+ * @brief  Default QDEC undefined value
+ */
+#ifndef QDEC_UNDEF
+#define QDEC_UNDEF           (UINT_MAX)
+#endif
+
+/**
+ * @brief   Default QDEC type definition
+ */
+#ifndef HAVE_QDEC_T
+typedef unsigned int qdec_t;
+#endif
+
+/**
+ * @brief   Default QDEC mode definition
+ */
+#ifndef HAVE_QDEC_MODE_T
+typedef enum {
+    QDEC_X1,    /* X1 mode */
+    QDEC_X2,    /* X2 mode */
+    QDEC_X4,    /* X4 mode */
+} qdec_mode_t;
+#endif
+
+/**
+ * @brief   Signature of event callback functions triggered from interrupts
+ *
+ * @param[in] arg       optional context for the callback
+ */
+typedef void (*qdec_cb_t)(void *arg);
+
+/**
+ * @brief   Default interrupt context entry holding callback and argument
+ */
+#ifndef HAVE_TIMER_ISR_CTX_T
+typedef struct {
+    qdec_cb_t cb;          /**< callback executed from qdec interrupt */
+    void *arg;              /**< optional argument given to that callback */
+} qdec_isr_ctx_t;
+#endif
+
+/**
+ * @brief   Initialize a QDEC device
+ *
+ * The QDEC module is based on virtual QDEC devices, which can have one or more
```

Comment:
This block of documentation was describing the pwm init function, but is not relevant here.
Since you already have explained a lot in the beginning of this file, you may just sum up a little and point toward the begining.

Reply:
Oh yes I forgot to correct that one. thanks.